src/variable.cpp

```diff
@@ -478,6 +467,14 @@ void Variable::removeInterfaceType()
     mPimpl->mInterfaceType.clear();
 }
 
+bool Variable::hasInterfaceType(InterfaceType interfaceType) const
+{
+    if (interfaceType == Variable::InterfaceType::NONE && mPimpl->mInterfaceType.empty()) {
+        return true;
+    }
+    return mPimpl->mInterfaceType == interfaceTypeToString.find(interfaceType)->second;
```

Comment:
I don't understand why we have an `enum` for the `interfaceType` if we store it as a string in the variable?  Why not go one way or the other?

Reply:
I don't have too much of a problem in offering both here.

Reply:
Why forbid it everywhere else then?  eg: generator profiles, issue kinds etc.  Should we add string overloads to those functions too?

Reply:
No, we shouldn't.  If anything we should remove the string from here.  This was a very early piece of code used for the very start where we were printing models hence why it is this way